Automattic/sensei
# Add installation error handler in the onboarding

Fixes #5649
Based on https://github.com/Automattic/sensei/pull/5847

### Changes proposed in this Pull Request

* It adds the installation error handler in the onboarding.
* It also adds a `clearAction` feature to the actions, so we can cancel it if the user is in the last step (there's a timer before the redirect), and the user goes back using the browser history. https://github.com/Automattic/sensei/pull/5867/commits/975e7e9b6fe4ea02a087caa7bd4bec06b678c943
* Notice that the feature wasn't implemented exactly as we planned. I think this way it's cleaner, and it's probably a good experience with the immediate feedback. WDYT?

### Testing instructions


* In the `Sensei_REST_API_Extensions_Controller::install_extension`, adds the following code to force an error:
```
return new WP_Error(
	'sensei_extensions_install_error',
	'Custom error message from the server.'
);
```
* Deactivate or uninstall WooCommerce and Sensei Certificates.
* Go through the setup wizard, and in the purpose step, select the 2 first options.
* Make sure that in the last step, it displays the error, and you can retry and skip it.
* While it's showing "Setting up your new Sensei Home", navigate back in your browser (to the newsletter step), and make sure it's not redirected to the home.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/194379627-36dd972e-2198-4b33-96dd-dd7c54d1ba75.mov
